bootloader/src/pyi_path.c

```diff
@@ -153,6 +154,7 @@ pyi_path_join(char *result, const char *path1, const char *path2)
 
     if (path2[len - 1] == PYI_SEP) {
         /* Append path2 without slash. */
+        *(result + len) = '\0';
```

Comment:
How does this avoid the warning?
For my understanding, this line should go behind the next one to *really* ensure the string is \o-terminated.